### PR TITLE
fix parameters_to_url_query returns booleans with upper letter

### PR DESCRIPTION
**Pull request by mhaley-tignis**
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by mhaley-tignis** [comment]
Fix for https://github.com/OpenAPITools/openapi-generator/issues/16941

**Comment by mhaley-tignis** [comment]
@cbornet @tomplus @krjakbrjak @fa0311 @multani

File: modules/openapi-generator/src/main/resources/python-pydantic-v1/api_client.mustache [hunk 1]
```diff
@@ -567,10 +567,10 @@ class ApiClient:
         if collection_formats is None:
             collection_formats = {}
         for k, v in params.items() if isinstance(params, dict) else params:  # noqa: E501
-            if isinstance(v, (int, float)):
-                v = str(v)
             if isinstance(v, bool):
                 v = str(v).lower()
+            if isinstance(v, (int, float)):
```
Comment (mhaley-tignis) [inline modules/openapi-generator/src/main/resources/python-pydantic-v1/api_client.mustache:572]: In python boolean is a subclass of int so `isinstance(True, (int, float)) == True`. We need to check if the type is boolean before checking if the type is a number to get the behaviour we desire.

**Comment by wing328** [comment]
@mhaley-tignis thanks for the PR. Can you please take a look at the test failure when you've time?

**Comment by mhaley-tignis** [comment]
> @mhaley-tignis thanks for the PR. Can you please take a look at the test failure when you've time?

whoops, yes, missed that `&`, see f6da1744c1010642d5c

**Comment by mhaley-tignis** [comment]
ok @wing328, should be good now. I rushed this and didn't really look at the exception the first time. Sorry about that!

**Comment by wing328** [comment]
no need to sorry and thanks for your PR to improve the Python client generator.

File: samples/client/echo_api/python-pydantic-v1/test/test_manual.py [hunk 2]
```diff
@@ -145,6 +145,11 @@ def testAuthHttpBasic(self):
         self.assertTrue("Authorization" in e.headers)
         self.assertEqual(e.headers["Authorization"], "Basic dGVzdF91c2VybmFtZTp0ZXN0X3Bhc3N3b3Jk")
 
+    def testParametersToUrlQuery(self):
```
Comment (multani) [inline samples/client/echo_api/python-pydantic-v1/test/test_manual.py:148]: @mhaley-tignis for consistency, can you rename this method and those in the other tests with the same name?

I think `test_parameters_to_url_query_boolean_value` is a good name and clearly explain the test :)

**Review by multani — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/resources/python-pydantic-v1/api_client.mustache (hunk 1 above)
Comment (multani) [reply modules/openapi-generator/src/main/resources/python-pydantic-v1/api_client.mustache:572]: > In python boolean is a subclass of int so `isinstance(True, (int, float)) == True`. We need to check if the type is boolean before checking if the type is a number to get the behaviour we desire.

TIL it's [still the case in Python 3](https://docs.python.org/3/library/stdtypes.html#boolean-type-bool), I thought this was only the case in 2.x 👍

**Review by multani — APPROVED** [review]
Looks good to me, thanks for the fix @mhaley-tignis !

File: samples/client/echo_api/python-pydantic-v1/test/test_manual.py (hunk 2 above)
Comment (mhaley-tignis) [reply samples/client/echo_api/python-pydantic-v1/test/test_manual.py]: I can, see f701d7daa50f1.  Most of the other tests in this file follow camel casing, want me to change them over too?

Comment (multani) [reply samples/client/echo_api/python-pydantic-v1/test/test_manual.py]: No, you can keep it this way I think.

**Pull request merged**